### PR TITLE
Adding more sorns

### PR DESCRIPTION
**Pull request by ondrae**
This PR:
- lets us add 302 more SORNs, based on what we learned in #10
- Sets up a dev server at https://allsornsdev.app.cloud.gov/search

It also refactors our code used to find and save SORNs, moving the logic out of the FIndSornsJob and into a FederalRegisterClient class, with reusable methods. This helps because we will need to add a small handful of SORNs by hand, and the new `FederalRegisterClient.new.add_sorn_by_url` method lets us do just that.

#### After Deploying
- ssh into the server
`cf ssh all_sorns_dev -t -c "/tmp/lifecycle/launcher /home/vcap/app 'rails c' ''"`
- Add SORNs by hand
`FederalRegisterClient.new.add_sorn_by_url('https://www.federalregister.gov/documents/2009/08/21/E9-20209/type-of-action-
mendment')`

title | url
-- | --
[No title available] | https://www.federalregister.gov/documents/2007/05/29/07-2638/no-title-available
[No title available] | https://www.federalregister.gov/documents/2004/03/15/04-5676/no-title-available
Amendment for the EPA Travel, Other Accounts Payable and Accounts Receivable Files (EPA-29) | https://www.federalregister.gov/documents/2015/07/30/2015-18722/amendment-for-the-epa-travel-other-accounts-payable-and-accounts-receivable-files-epa-29
Amendment of Inspector General Operations & Reporting System Audit, Assignment, and Timesheet Files (EPA-42) | https://www.federalregister.gov/documents/2011/07/19/2011-18158/amendment-of-inspector-general-operations-and-reporting-system-audit-assignment-and-timesheet-files
Amendment of Inspector General's Operation and Reporting (IGOR) System Investigative Files (EPA-40) | https://www.federalregister.gov/documents/2011/11/16/2011-29655/amendment-of-inspector-generals-operation-and-reporting-igor-system-investigative-files-epa-40
Amendment of OIG Hotline Allegation System (EPA-30) | https://www.federalregister.gov/documents/2011/07/19/2011-18161/amendment-of-oig-hotline-allegation-system-epa-30
Amendment of PeoplePlus (EPA-1) | https://www.federalregister.gov/documents/2014/06/05/2014-13058/amendment-of-peopleplus-epa-1
Amendment of System Records Notice for the PeoplePlus | https://www.federalregister.gov/documents/2007/07/09/E7-13205/amendment-of-system-records-notice-for-the-peopleplus
Amendment of the External Compliance Program Discrimination Complaint Files, EPA-21 | https://www.federalregister.gov/documents/2014/10/24/2014-25417/amendment-of-the-external-compliance-program-discrimination-complaint-files-epa-21
Amendment of the Federal Docket Management System (EPA/GOV-2) | https://www.federalregister.gov/documents/2014/02/18/2014-03430/amendment-of-the-federal-docket-management-system-epagov-2
Amendment of the Federal Docket Management System (EPA/GOV-2) | https://www.federalregister.gov/documents/2013/10/02/2013-24120/amendment-of-the-federal-docket-management-system-epagov-2
Amendment to General Routine Uses | https://www.federalregister.gov/documents/2008/01/14/E8-445/amendment-to-general-routine-uses
Amendment to the Toxic Substances Control Act Confidential Business Information Records Access System, EPA-20 | https://www.federalregister.gov/documents/2009/04/07/E9-7818/amendment-to-the-toxic-substances-control-act-confidential-business-information-records-access
Type of Action: Amendment | https://www.federalregister.gov/documents/2009/08/21/E9-20209/type-of-action-amendment

### CODE REVIEW

**Comment by ondrae** [comment]
We got 243 more SORNs with this code on our dev server. https://allsornsdev.app.cloud.gov/search

**Comment by ondrae** [comment]
Two questions I have about this PR:
- Does FederalRegisterApi and its methods make sense?
- Do we need a task to call .add_sorn_by_url or is sshing into the machine okay?

File: .cloud-gov/manifest-dev.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
```
Comment (lauraGgit) [inline .cloud-gov/manifest-dev.yml:1]: per this, were you hoping to have this push via CI/CD in a certain tagged branch or you're still planning on manually deploying to the dev instance?

File: app/jobs/find_sorns_job.rb [hunk 2]
```diff
@@ -2,86 +2,6 @@ class FindSornsJob < ApplicationJob
...
-      conditions: conditions,
-      type: 'NOTICE', # doesn't seem to work
-      fields: fields,
-      order: 'newest', #oldest
-      per_page: 200,
-      page: 1
-    }
-
-    search_fed_reg(search_options)
+    FederalRegisterApi.find_sorns
```
Comment (lauraGgit) [inline app/jobs/find_sorns_job.rb:5]: was i under the correct impression that the FR had its own maintained API or gem? if that's the case would added something like `Client` help differentiate it?

File: app/models/federal_register_api.rb [hunk 3]
```diff
@@ -0,0 +1,97 @@
+class FederalRegisterApi
+  def self.find_sorns
+    # Makes queries to the Federal Register API to find SORNs.
+    # Searching for 'Privacy Act of 1974; System of Records' of type 'Notice' is the best query we have found.
+    # In the results we still filter on those with a title that includes 'Privacy Act of 1974'.
+
+    conditions = { term: 'Privacy Act of 1974; System of Records' }#, agencies: ['general-services-administration'] }
```
Comment (lauraGgit) [inline app/models/federal_register_api.rb:7]: would it make sense to put the `conditions`, `fields` and `search options` in an class constructor?

**Review by lauraGgit — COMMENTED** [review]
While conceptually, I like the refactor, I'm not exactly sure how the refactor adds 302 more SORNs.

File: .cloud-gov/manifest-dev.yml (hunk 1 above)
Comment (ondrae) [reply .cloud-gov/manifest-dev.yml:1]: For now, I will just push manually to this dev server when we want to.

I'd like to get to the place where all PRs get their own deployment automatically. Greg said he's done that before with cloud.gov. The PR servers are great for sharing ideas and work in progress.

File: app/jobs/find_sorns_job.rb (hunk 2 above)
Comment (ondrae) [reply app/jobs/find_sorns_job.rb:5]: Yeah, any name suggestions are welcome. `Client` or `Wrapper` are better than what I have.

**Comment by ondrae** [comment]
> While conceptually, I like the refactor, I'm not exactly sure how the refactor adds 302 more SORNs.

This [first commit](https://github.com/18F/all_sorns/commit/ddee485f10d1e1b7ba51077a5686bfeac5007c7e) adds a few more accepted SORN titles. Which is most of what we were missing. 

We still needed to add a handful by hand though, which what the rest of the commits and the refactor let us do.

File: app/jobs/find_sorns_job.rb (hunk 2 above)
Comment (ondrae) [reply app/jobs/find_sorns_job.rb]: Changed to FederalRegisterClient.

File: app/models/federal_register_api.rb (hunk 3 above)
Comment (ondrae) [reply app/models/federal_register_api.rb]: Changed this now use instances of the class. It'll be easier to make passing in variables now, if we ever need to add that ability.

**Review by lauraGgit — APPROVED** [review]
LGTM 🚀

**Pull request merged**